rust/rope/src/delta.rs

```diff
@@ -341,6 +336,60 @@ impl<'a, N: NodeInfo + 'a> Transformer<'a, N> {
     }
 }
 
+/// A builder for creating new `Delta` objects.
+///
+/// Note that all edit operations must be sorted; the start point of each
+/// interval must be no less than the end point of the previous one.
+pub struct Builder<N: NodeInfo> {
+    delta: Delta<N>,
+    last_offset: usize,
+}
+
+impl<N: NodeInfo> Builder<N> {
+    /// Creates a new builder, applicable to a base rope of length `base_len`.
+    pub fn new(base_len: usize) -> Builder<N> {
+        Builder {
+            delta: Delta {
+                els: Vec::new(),
+                base_len: base_len,
+            },
+            last_offset: 0,
+        }
+    }
+
+    /// Deletes the given interval. Panics if interval is not properly sorted.
+    pub fn delete(&mut self, interval: Interval) {
+        let (start, end) = interval.start_end();
+        if start < self.last_offset {
+            panic!("Delta builder: intervals not properly sorted");
+        }
+        if start > self.last_offset {
+            self.delta.els.push(DeltaElement::Copy(self.last_offset, start));
+        }
+        self.last_offset = end;
+    }
+
+    /// Replaces the given interval with the new rope. Panics if interval
+    /// is not properly sorted.
+    pub fn replace(&mut self, interval: Interval, rope: Node<N>) {
+        self.delete(interval);
+        self.delta.els.push(DeltaElement::Insert(rope));
+    }
+
+    /// Determines if delta would be trivial if built
+    pub fn is_trivial(&self) -> bool {
```

Comment:
As similarly suggested in #256, it may be more clear to name this one `is_empty`